packages/devtools_app/lib/src/analytics/analytics.dart

```diff
@@ -86,14 +89,69 @@ class GtagEventDevTools extends GtagEvent {
     String ide_launched, // dimension7 Devtools launched (CLI, VSCode, Android)
     String flutter_client_id, // dimension8 Flutter tool client_id (~/.flutter).
 
-    int raster_duration_micros,
+    // Performance screen metrics. See [PerformanceScreenMetrics].
     int ui_duration_micros,
+    int raster_duration_micros,
     int shader_compilation_duration_micros,
+    int trace_event_count,
+    // Profiler screen metrics. See [ProfilerScreenMetrics].
     int cpu_sample_count,
     int cpu_stack_depth,
-    int trace_event_count,
   });
 
+  factory GtagEventDevTools.withScreenMetrics({
+    String event_category,
+    String event_label,
+    String send_to,
+    int value,
+    bool non_interaction,
+    dynamic custom_map,
+    String user_app,
+    String user_build,
+    String user_platform,
+    String devtools_platform,
+    String devtools_chrome,
+    String devtools_version,
+    String ide_launched,
+    String flutter_client_id,
+    ScreenAnalyticsMetrics screenMetrics,
+  }) {
+    return GtagEventDevTools(
+      event_category: event_category,
+      event_label: event_label,
+      send_to: send_to,
+      non_interaction: non_interaction,
+      value: value,
+      user_app: userAppType,
+      user_build: userBuildType,
+      user_platform: userPlatformType,
+      devtools_platform: devtoolsPlatformType,
+      devtools_chrome: devtoolsChrome,
+      devtools_version: devtoolsVersion,
+      ide_launched: ideLaunched,
+      flutter_client_id: flutterClientId,
+      ui_duration_micros: screenMetrics is PerformanceScreenMetrics
```

Comment:
having is checks for subtypes isn't very robust. Can you instead switch to adding a toJson() or similar method in the ScreenAnalyticsMetrics subclasses that surfaces the custom metrics each screen cares about?
Will requiring tweaking GtagEventDevTools to take a JSON map of extra parameters rather than just a fixed list of fields but probably worth it to keep adding extra metrics simple.

Reply:
To land this now add a TODO here  and a comment in the ScreenAnalyticsMetrics base class clarifying that extending this class is fragile.

Reply:
I thought about this. Since the `GtagEventDevTools` is an external factory constructor, I'm not sure how passing json in this would affect the custom dimension setup in JS.

I will go ahead and add a TODO to consider trying to make this take a json map in the future, but for now I will add better documentation.